Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/question/GroupingQuestion.kt

```diff
@@ -0,0 +1,281 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.compose.jetsurvey.survey.question
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringArrayResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.example.compose.jetsurvey.R
+import com.example.compose.jetsurvey.survey.Answer
+import com.example.compose.jetsurvey.survey.PossibleAnswer
+import com.example.compose.jetsurvey.survey.question.components.DragTarget
+import com.example.compose.jetsurvey.survey.question.components.Draggable
+import com.example.compose.jetsurvey.survey.question.components.DropZone
+import com.example.compose.jetsurvey.survey.question.components.LocalDragTargetInfo
+import com.example.compose.jetsurvey.survey.question.components.OverflowContainer
+
+const val STARTING_ZONE_BADGE_GROUP = "Placeholder"
+
+@Composable
+fun GroupingQuestion(
+    possibleAnswer: PossibleAnswer.Group,
+    answer: Answer.Group?,
+    onAnswerReady: (Map<String, List<Badge>>) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val initialBadgeList =
+        stringArrayResource(possibleAnswer.items).mapIndexed { index, item ->
+            Badge(index, item)
+        }
+    val badgeGroupList =
+        stringArrayResource(possibleAnswer.groups)
+            // Make sure no group name uses reserved name
+            .filter { it != STARTING_ZONE_BADGE_GROUP }
+            .distinct()
+    var groupAnswer by remember {
```

Comment:
Consider using `rememberSaveable` so user doesn't lose their progress on configuration changes.

Reply:
Added in the next commit!